apps/condo/domains/common/components/TrackingContext.tsx

```diff
@@ -105,35 +110,48 @@ const useTracking: IUseTracking = () => {
         return fn as T
     }
 
-    const getEventName = (eventType: TrackingEventType) => {
-        const [domainName, isDetail, suffix] = compact(route.split('/'))
+    const getEventName: IGetEventName = (eventType) => {
+        const [domainName, detailPageName, suffix, customPage] = compact(route.split('/'))
         if (!domainName) {
             return ''
         }
-        const domain = domainName.charAt(0).toUpperCase() + domainName.slice(1)
+        const domain = upperFirst(domainName)
         let domainSuffix = 'Index'
         let domainPostfix = ''
 
-        switch (isDetail) {
+        switch (detailPageName) {
             case '[id]':
                 domainSuffix = 'Detail'
                 break
             case 'create':
-                domainSuffix = 'Create'
+            case 'update':
+            case 'hint':
+            case 'detail':
+            case 'signin':
+            case 'register':
+            case 'change-password':
+            case 'forgot':
+                domainSuffix = upperFirst(detailPageName)
                 break
+
         }
 
-        if (suffix) {
-            switch (suffix) {
+        const postfixToken = customPage ? customPage : suffix
+
+        if (postfixToken) {
+            switch (postfixToken) {
                 case 'update':
                     domainPostfix = 'Update'
                     break
                 case 'pdf':
                     domainPostfix = 'Pdf'
                     break
+                case '[id]':
+                    domainSuffix = 'Detail'
+                    break
                 default:
                     // convert string from domain-related-page to DomainRelatedPage
-                    domainPostfix = suffix.replace(/(-\w|^\w)/g, (_, g) => g.slice(-1).toUpperCase())
+                    domainPostfix = postfixToken.replace(/(-\w|^\w)/g, (_, g) => g.slice(-1).toUpperCase())
```

Comment:
is it possible to put regex in a constant and give it a clear name instead of a comment?